libcalico-go/lib/backend/k8s/resources/ipam_config.go

```diff
@@ -113,6 +167,12 @@ func (c *ipamConfigClient) Create(ctx context.Context, kvp *model.KVPair) (*mode
 	if err != nil {
```

Comment:
Here's an example - if `kvp` is already a v3 object, then `c.toV3` is a no-op right?

I think this code is getting a little bit (more) confusing with this change, since a few things are not clear:

- What does v3 mean? Is it "projectcalico.org/v3", "crd.projectcalico.org/v1", or both? 
- What is the expected input to this function? Looks like it can take multiple formulations of the data now.

I think at a minimum we should clearly document the expected types of data, but potentially we should have separate functions to handle this and split it:

```
func (c *client) Create(ctx, context.Context, kvp *model.KVPair) (kvp, error) {
  switch kvp {
    case fromIPAMCode:
      // From the IPAM code - we need to convert to CRD format.
      return c.createV1()
    case fromClientV3:
      // From the v3 client - it's already in CRD format.
      return c.createV3()
    default:
      panic("Bad type!") 
  }
}
```

Reply:
- v3 means the KVP is passed from `libcalico-go/lib/clientv3` with the format:
key: model.ResourceKey
value: *libapiv3.IPAMConfig

- v1 means the KVP is passed from IPAM code with the format (I presume it is the original format used by v1 backend client? )
key: model.IPAMConfigKey
value: *model.IPAMConfig

I think using separate functions is a much clearer approach. I'll update the code.